src/core/network/qgsnetworkcontentfetcherregistry.cpp

```diff
@@ -163,7 +167,12 @@ void QgsFetchedContent::taskCompleted()
     QNetworkReply *reply = mFetchingTask->reply();
     if ( reply->error() == QNetworkReply::NoError )
     {
-      QTemporaryFile *tf = new QTemporaryFile( QStringLiteral( "XXXXXX" ) );
+      // keep extension, it can be useful when guessing file content
+      // (when loading this file in a Qt WebView for instance)
+      const QString extension = QFileInfo( reply->request().url().fileName() ).completeSuffix();
```

Comment:
should you check that the extension does exist to avoid a temp file ending with a point, or we don't care?

Reply:
It's done the line after, if extension is empty, we fallback on classic temporary filepath name

Reply:
oops, thanks